src/snowflake/connector/connection.py

```diff
@@ -1115,7 +1135,12 @@ def __config(self, **kwargs):
             ]:
                 self._authenticator = auth_tmp
 
-        if not self.user and self._authenticator != OAUTH_AUTHENTICATOR:
+        if (
+            not self._master_token
```

Comment:
I feel like instead of modifying this and the next if statements it'd be better to rearrange the code so that error checking (btw you missed the account one) could simply be indented with the other old code.

Reply:
Okay, rearranged! Let me know if that's what you were suggesting